### PR TITLE
Epoll android

### PR DESCRIPTION
**Pull request by 4ntoine**
Motivation:

This can be a pretty rare use case, but i'd like to have gRPC service written in C++ and gRPC client running in Java (Android). The main motivation is to use gRPC as JNI alternative on Android.
[Here](https://github.com/4ntoine/JniBenchmarking/blob/a056d4030fc0d9841c48d62f6209b2ca773a18d5/rpc/src/androidTest/java/com/eyeo/ctu/rpc/AndroidRpcTest.kt#L166) is the test.

The main issue is that platform binaries have to be compiled for Android. Also i've found [some Android incompatibility](https://github.com/4ntoine/netty/issues/1) and fixed it.

Modification:

* added Gradle build configuration files (both native part and managed part)
* addede "netty-jni-util" as Git submodule.
* `FileDescriptor.fd` is falled back to `FileDescriptor.descriptor` on Android (i believe it still works for desktop).
* some minor changes in Git-ignore file
* bundled Gradle scripts (typical for Android projects).

Result:

If there is no issue then describe the changes introduced by this PR.
in the root directory one can ru `./gradlew publishToMavenLocal` and use `'io.netty:netty-transport-native-epoll-android:4.1.55.Final@aar'` in his Android project. Note some other deps are still required.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "netty-jni-util"]
```
Comment (4ntoine) [inline .gitmodules:1]: This is less-likely to be desired. What's a better way to do it?

File: gradlew [hunk 2]
```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env sh
```
Comment (4ntoine) [inline gradlew:1]: typical wrapper file

File: gradlew.bat [hunk 3]
```diff
@@ -0,0 +1,84 @@
+@if "%DEBUG%" == "" @echo off
```
Comment (4ntoine) [inline gradlew.bat:1]: typical wrapper file

File: transport-native-epoll/build.gradle [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      path 'CMakeLists.txt'
+    }
+  }
+}
+
+dependencies {
+  api 'io.netty:netty-transport-native-epoll:4.1.55.Final'
+}
+
+// `./gradlew publishToMavenLocal` to publish
```
Comment (4ntoine) [inline transport-native-epoll/build.gradle:36]: this is required to be done manually before usage. Also it requires adding `mavemLocal()` to consumer project. As [a probably better alternative] one can add CI build config to build it with Gradle (after Maven) or update Maven build configuration

File: transport-native-epoll/build.gradle [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+publishing {
+    publications {
+        transport_native_epoll(MavenPublication) {
+            artifact("$buildDir/outputs/aar/transport-native-epoll-release.aar") {
+              group 'io.netty'
+              artifactId 'netty-transport-native-epoll-android'
```
Comment (4ntoine) [inline transport-native-epoll/build.gradle:43]: ideally i'd use `classifier` here but it was not working when consuming with Gradle later.

File: transport-native-epoll/build.gradle [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  }
+  externalNativeBuild {
+    cmake {
+      path 'CMakeLists.txt'
+    }
+  }
+}
+
+dependencies {
+  api 'io.netty:netty-transport-native-epoll:4.1.55.Final'
```
Comment (4ntoine) [inline transport-native-epoll/build.gradle:33]: just to include java classes.

**Comment by normanmaurer** [comment]
Honestly I think that is not something we want to support... I mean I am happy to take the c file fixes but adding more build config etc sounds like something we don't want to maintain.

**Comment by 4ntoine** [comment]
@normanmaurer Hi, Norman. That's fine with me, feel free to get what you're interested in.

**Comment by normanmaurer** [comment]
@4ntoine https://github.com/netty/netty/pull/10882

**Pull request closed without merging**